Loop/Managers/DeviceDataManager.swift

```diff
@@ -212,45 +222,69 @@ class DeviceDataManager: CarbStoreDelegate, TransmitterDelegate {
             }
         }
     }
-
+    
     private func fetchPumpHistory() {
+        
         guard let device = rileyLinkManager.firstConnectedDevice else {
             return
         }
-
+        
         /*
          This is where we fetch history.
-        */
-
-        let startDate = doseStore.pumpEventQueryAfterDate
-
+         */
+        
+        // TODO: Reconcile these
+        //let startDate = doseStore.pumpEventQueryAfterDate
+        let startDate = nightscoutUploader?.observingPumpEventsSince ?? observingPumpEventsSince
+        
         device.ops?.getHistoryEventsSinceDate(startDate) { (result) in
             switch result {
-            case let .Success(events, _):
+            case let .Success(events, pumpModel):
```

Comment:
I've changed this in a coming PR, since `pumpModel` ends up just returning the value set to `device.ops?.pumpState.pumpModel`